test/integration_test_helper.rb

```diff
@@ -20,18 +20,40 @@
 end
 
 Selenium::WebDriver::Chrome::Service.driver_path = ENV['TESTDRIVER_PATH'] || Foreman::Util.which('chromedriver', Rails.root.join('node_modules', '.bin'))
+
+options = Selenium::WebDriver::Chrome::Options.new
+options.args << '--disable-gpu'
+options.args << '--no-sandbox'
+options.args << '--window-size=1024,768'
+options.args << '--headless' unless ENV['DEBUG_JS_TEST'] == '1'
+
+selenium_remote_host = ENV.fetch('SELENIUM_REMOTE_HOST', nil)
+selenium_remote_port = ENV.fetch('SELENIUM_REMOTE_PORT', 4444)
+
 Capybara.register_driver :selenium_chrome do |app|
-  options = Selenium::WebDriver::Chrome::Options.new
-  options.args << '--disable-gpu'
-  options.args << '--no-sandbox'
-  options.args << '--window-size=1024,768'
-  options.args << '--headless' unless ENV['DEBUG_JS_TEST'] == '1'
-  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options)
+  if selenium_remote_host
+    Capybara::Selenium::Driver.new(
+      app,
+      browser: :remote,
+      url: "http://#{selenium_remote_host}:#{selenium_remote_port}/wd/hub",
+      options: options)
+  else
+    Capybara::Selenium::Driver.new(
+      app,
+      browser: :chrome,
+      options: options)
+  end
 end
+
 Capybara.configure do |config|
-  config.javascript_driver      = ENV["JS_TEST_DRIVER"]&.to_sym || :selenium_chrome
-  config.default_max_wait_time  = 20
+  config.javascript_driver = ENV["JS_TEST_DRIVER"]&.to_sym || :selenium_chrome
+  config.default_max_wait_time = 20
   config.enable_aria_label = true
+  if selenium_remote_host
+    config.server_host = Socket.ip_address_list
+                           .find(&:ipv4_private?)
+                           .ip_address
```

Comment:
What's the logic behind this? Isn't `::` (all IPs) a better choice?
```suggestion
    config.server_host = '::'
```

Reply:
It looks like ipv6 is currently not possible with capybara or selenium:
```
 {"url":"http://:::41231/users/login"}
2023-12-20 08:47:01 INFO Selenium <- {"value":{"error":"invalid argument","message":"invalid argument\n  (Session info: headless chrome=119.0.6045.199)"
```
But we could try with `0.0.0.0`?

Reply:
It may work with [::] but otherwise 0.0.0.0 probably works too. Fixing capybara to escape IPv6 would be a good thing